src/components/views/settings/SetIdServer.js

```diff
@@ -179,73 +204,80 @@ export default class SetIdServer extends React.Component {
             checking: false,
             error: errStr,
             currentClientIdServer: MatrixClientPeg.get().getIdentityServerUrl(),
-            idServer: this.state.idServer,
         });
     };
 
     _showNoTermsWarning(fullUrl) {
         const QuestionDialog = sdk.getComponent("views.dialogs.QuestionDialog");
-        Modal.createTrackedDialog('No Terms Warning', '', QuestionDialog, {
-            title: _t("Identity server has no terms of service"),
-            description: (
-                <div>
-                    <span className="warning">
-                        {_t("The identity server you have chosen does not have any terms of service.")}
-                    </span>
-                    <span>
-                        &nbsp;{_t("Only continue if you trust the owner of the server.")}
-                    </span>
-                </div>
-            ),
-            button: _t("Continue"),
-            onFinished: async (confirmed) => {
-                if (!confirmed) return;
-                this._saveIdServer(fullUrl);
-            },
+        return new Promise(resolve => {
+            Modal.createTrackedDialog('No Terms Warning', '', QuestionDialog, {
```

Comment:
thanks to t3chguy this now returns access to a thenable - we should use that

Reply:
Changed to this format, thanks!